Add May 28 Elasticsearch serverless release notes (#6597)
elastic/docs-content: release-notes/elastic-cloud-serverless/_snippets/2026-05-20/index.md

﻿## May 20, 2026 [elastic-release-notes-2026-05-20]
### Features and enhancements [elastic-2026-05-20-features-enhancements]
* Add per-tier `deletion_costs` to autoscaling metrics API.
* Reserve circuit breaker memory for automaton construction. [#148722](https://github.com/elastic/elasticsearch/pull/148722) [#147428](https://github.com/elastic/elasticsearch/issues/147428)
* Improve ES|QL field extraction on flattened fields. [#148891](https://github.com/elastic/elasticsearch/pull/148891)

> * Add query approximation support for ES|QL queries containing FORK. [#148412](https://github.com/elastic/elasticsearch/pull/148412)

Reviewer: ```suggestion
* Add query approximation support for ES|QL queries containing `FORK`. [#148412](https://github.com/elastic/elasticsearch/pull/148412)
```

Author: Fixed in https://github.com/elastic/elasticsearch-serverless/pull/6731

* Implement bucket metadata. [#146077](https://github.com/elastic/elasticsearch/pull/146077)

> * Add default schema resolution to UNION_BY_NAME. [#149176](https://github.com/elastic/elasticsearch/pull/149176)

Reviewer: ```suggestion
* Add default schema resolution to `UNION_BY_NAME`. [#149176](https://github.com/elastic/elasticsearch/pull/149176)
```

Author: Fixed in https://github.com/elastic/elasticsearch-serverless/pull/6731

> * Speed up REPLACE on constant regex. [#149033](https://github.com/elastic/elasticsearch/pull/149033)

Reviewer: ```suggestion
* Speed up `REPLACE` on constant regex. [#149033](https://github.com/elastic/elasticsearch/pull/149033)
```

Author: Fixed in https://github.com/elastic/elasticsearch-serverless/pull/6731

* Cut S3 download + Snappy copies on hot path. [#149248](https://github.com/elastic/elasticsearch/pull/149248)
* Improve ES|QL Parquet footer cache reliability. [#149045](https://github.com/elastic/elasticsearch/pull/149045)

> * Add logging tracking to _xpack/usage. [#148087](https://github.com/elastic/elasticsearch/pull/148087)

Reviewer: ```suggestion
* Add logging tracking to the `/_xpack/usage` endpoint. [#148087](https://github.com/elastic/elasticsearch/pull/148087)
```

Author: Fixed in https://github.com/elastic/elasticsearch-serverless/pull/6731

* Improve ES|QL packed-values grouping performance. [#148681](https://github.com/elastic/elasticsearch/pull/148681)
* Improve S3 receive buffer pooling for ES|QL external sources. [#149023](https://github.com/elastic/elasticsearch/pull/149023)
* Improve ES|QL TopN field extraction for external sources. [#149185](https://github.com/elastic/elasticsearch/pull/149185)
* Improve ES|QL Parquet and ORC footer caching. [#149018](https://github.com/elastic/elasticsearch/pull/149018)

> * Improve performance of REPLACE fast-path in ES|QL. [#149167](https://github.com/elastic/elasticsearch/pull/149167)

Reviewer: ```suggestion
* Improve performance of `REPLACE` fast-path in ES|QL. [#149167](https://github.com/elastic/elasticsearch/pull/149167)
```

Author: Fixed in https://github.com/elastic/elasticsearch-serverless/pull/6731

* Improve ES|QL TopN UTF-8 string encoding performance. [#149022](https://github.com/elastic/elasticsearch/pull/149022)

> * Improve performance of Parquet datasource with nested-And short-circuit and filter eval cleanups. [#148935](https://github.com/elastic/elasticsearch/pull/148935)

Reviewer: ```suggestion
* Improve performance of Parquet datasource with nested-And short-circuit and clean up filter evaluation. [#148935](https://github.com/elastic/elasticsearch/pull/148935)
```

* Improve reporting of external source rows in ES|QL metrics. [#148410](https://github.com/elastic/elasticsearch/pull/148410)
* Improve performance of pushed-down filter evaluation in the ES|QL Parquet datasource. [#148918](https://github.com/elastic/elasticsearch/pull/148918)

> * Add SIMD contains for literal LIKE patterns in ES|QL. [#149026](https://github.com/elastic/elasticsearch/pull/149026)

Reviewer: ```suggestion
* Add SIMD `contains` for literal `LIKE` patterns in ES|QL. [#149026](https://github.com/elastic/elasticsearch/pull/149026)
```

* Add an ES|QL optimizer rule that rewrites `SUM(X+c)` to `SUM(X) + c*COUNT(X)`. [#145510](https://github.com/elastic/elasticsearch/pull/145510) [#140470](https://github.com/elastic/elasticsearch/issues/140470)
* Add generic byte matchers for pushdown predicates in ES|QL. [#149030](https://github.com/elastic/elasticsearch/pull/149030)
* Reduce memory used by translog blob replication. [#148881](https://github.com/elastic/elasticsearch/pull/148881)
* Reduce translog memory when active shards have zero operations. [#148992](https://github.com/elastic/elasticsearch/pull/148992)
* Add S3 storage class tracking in object storage tests. [#149069](https://github.com/elastic/elasticsearch/pull/149069)

> * Add TS_COLLAPSE command. [#148690](https://github.com/elastic/elasticsearch/pull/148690)

Reviewer: ```suggestion
* Add `TS_COLLAPSE` command. [#148690](https://github.com/elastic/elasticsearch/pull/148690)
```

> * Reindex parse exception has document index and ID. [#145146](https://github.com/elastic/elasticsearch/pull/145146) [#48432](https://github.com/elastic/elasticsearch/issues/48432)

Reviewer: ```suggestion
* Add document index and ID to reindex parse exception messages. [#145146](https://github.com/elastic/elasticsearch/pull/145146) [#48432](https://github.com/elastic/elasticsearch/issues/48432)
```

* Enable stateless snapshots and allow relocation.
* Reject reindexes with wrong parameters. [#145290](https://github.com/elastic/elasticsearch/pull/145290) [#37855](https://github.com/elastic/elasticsearch/issues/37855)

> * Decrease tsdb `look_ahead_time` to 9 minutes. [#145552](https://github.com/elastic/elasticsearch/pull/145552) [#142602](https://github.com/elastic/elasticsearch/issues/142602)

Reviewer: ```suggestion
* Decrease TSDB `look_ahead_time` to 9 minutes. [#145552](https://github.com/elastic/elasticsearch/pull/145552) [#142602](https://github.com/elastic/elasticsearch/issues/142602)
```

> * Support jina-embeddings-v5-omni input format for Jina integration. [#148551](https://github.com/elastic/elasticsearch/pull/148551)

Reviewer: ```suggestion
* Support `jina-embeddings-v5-omni` input format for Jina integration. [#148551](https://github.com/elastic/elasticsearch/pull/148551)
```

* Add product use case and origin to APM telemetry. [#148140](https://github.com/elastic/elasticsearch/pull/148140)
* Add AI bots to user-agent regex. [#148665](https://github.com/elastic/elasticsearch/pull/148665)
* Improve field metadata caching per shard for mapping. [#148865](https://github.com/elastic/elasticsearch/pull/148865)
* Assign downsampling to least loaded eligible node. [#148220](https://github.com/elastic/elasticsearch/pull/148220)

### Fixes [elastic-2026-05-20-fixes]
* Fix missing deprecation warnings in transform create and preview APIs. [#147934](https://github.com/elastic/elasticsearch/pull/147934) [#82936](https://github.com/elastic/elasticsearch/issues/82936) [#82935](https://github.com/elastic/elasticsearch/issues/82935)
* Fix centroid grouping failures in ES|QL aggregations. [#148815](https://github.com/elastic/elasticsearch/pull/148815) [#141318](https://github.com/elastic/elasticsearch/issues/141318)

> * Fix wrong warning in ES|QL STATS for unrolled multivalued fields. [#145968](https://github.com/elastic/elasticsearch/pull/145968)

Reviewer: ```suggestion
* Fix wrong warning in ES|QL `STATS` for unrolled multivalued fields. [#145968](https://github.com/elastic/elasticsearch/pull/145968)
```

Author: Fixed in https://github.com/elastic/elasticsearch-serverless/pull/6731

* Fix ES|QL "optimized incorrectly" error from project reorder. [#149053](https://github.com/elastic/elasticsearch/pull/149053) [#148612](https://github.com/elastic/elasticsearch/issues/148612)
* Fix ES|QL Parquet pushdown for non-DOUBLE physical columns. [#149100](https://github.com/elastic/elasticsearch/pull/149100)
* Fix stalled ES|QL reads from CSV and TSV sources. [#149189](https://github.com/elastic/elasticsearch/pull/149189)
* Fix ES|QL histogram errors when filtering null values. [#148983](https://github.com/elastic/elasticsearch/pull/148983) [#147854](https://github.com/elastic/elasticsearch/issues/147854)

> * Fix UNION BY NAME COUNT/MIN/MAX pushdown for absent columns. [#149279](https://github.com/elastic/elasticsearch/pull/149279)

Reviewer: ```suggestion
* Fix `UNION BY NAME` COUNT/MIN/MAX pushdown for absent columns. [#149279](https://github.com/elastic/elasticsearch/pull/149279)
```

Author: Changed to `UNION_BY_NAME` in the changelogs.

> * Fix unclear ES|QL error for SPARKLINE after TS command. [#149015](https://github.com/elastic/elasticsearch/pull/149015)

Reviewer: ```suggestion
* Fix unclear ES|QL error for `SPARKLINE` after `TS` command. [#149015](https://github.com/elastic/elasticsearch/pull/149015)
```

Author: Fixed in https://github.com/elastic/elasticsearch-serverless/pull/6731

* Fix ES|QL type errors on multi-file CSV and NDJSON data sources. [#148756](https://github.com/elastic/elasticsearch/pull/148756)
* Fix ES|QL coordination thread pool usage. [#149034](https://github.com/elastic/elasticsearch/pull/149034)
* Fix premature release of ES|QL search contexts on failures. [#145960](https://github.com/elastic/elasticsearch/pull/145960) [#145509](https://github.com/elastic/elasticsearch/issues/145509)
* Fix ES|QL hangs on multi-file compressed external data sources. [#148797](https://github.com/elastic/elasticsearch/pull/148797)
* Fix ES|QL Parquet NaN handling in column-index statistics. [#149094](https://github.com/elastic/elasticsearch/pull/149094)
* Fix ES|QL null handling after JVM workaround. [#149050](https://github.com/elastic/elasticsearch/pull/149050)

> * Fix ES|QL sorting on histogram and tsid fields. [#149085](https://github.com/elastic/elasticsearch/pull/149085)

Reviewer: ```suggestion
* Fix ES|QL sorting on `histogram` and `_tsid` fields. [#149085](https://github.com/elastic/elasticsearch/pull/149085)
```

Author: Fixed in https://github.com/elastic/elasticsearch-serverless/pull/6731

> * Fix ? wildcard matching in `constant_keyword` wildcard queries. [#148585](https://github.com/elastic/elasticsearch/pull/148585) [#141785](https://github.com/elastic/elasticsearch/issues/141785)

Reviewer: ```suggestion
* Fix `?` wildcard matching in `constant_keyword` wildcard queries. [#148585](https://github.com/elastic/elasticsearch/pull/148585) [#141785](https://github.com/elastic/elasticsearch/issues/141785)
```

Author: Fixed in https://github.com/elastic/elasticsearch-serverless/pull/6731
